src/odemis/util/transform.py

```diff
@@ -886,11 +903,11 @@ def __init__(self, coeffs=None, rotation=None, scale=None, shear=None,
         elif coeffs is not None:
             self.coeffs = coeffs
         else:
-            self.rotation = 0. if rotation is None else rotation
-            self.scale = (1., 1.) if scale is None else scale
-            self.shear = 0. if shear is None else shear
+            GeometricTransform.__init__(self, rotation=rotation, scale=scale,
+                                        shear=shear, translation=translation)
             self.nlcoeffs = (0., 0., 0., 0., 0., 0., 0.) if nlcoeffs is None else nlcoeffs
-            self.translation = (0., 0.) if translation is None else translation
+            if len(self.nlcoeffs) != 7 or not all(isinstance(a, numbers.Real) for a in self.nlcoeffs):
+                raise ValueError("Scale should be 7 floats, but got %s" % (self.nlcoeffs,))
```

Comment:
Copy paste error? I think "Scale ....." should be "Coefficients ...." .

Reply:
Yep